### PR TITLE
Arc-Script Updates

### PR DESCRIPTION
**Pull request by segeljakt**

### CODE REVIEW

**Comment by segeljakt** [comment]
this was working on my machine before I uploaded it 👀

**Comment by segeljakt** [comment]
ok, now I see what can go wrong with `compiletest_rs`. If you got duplicate build artifacts (`cargo check` and `cargo test`), then `compiletest_rs` cannot distinguish between them and might pick the wrong one (`cargo check`) by chance

the solution is to do `cargo clean` before `cargo test`

**Review by frej — APPROVED** [review]
LGTM if you figure out why it's crashing :)

**Comment by segeljakt** [comment]
I am 99% sure this will work, let's see

File: arc-mlir/src/CMakeLists.txt [hunk 1]
```diff
@@ -25,4 +25,6 @@ get_filename_component(ARC_SCRIPT_SRC_DIR "../../arc-script"
   REALPATH BASE_DIR ${CMAKE_CURRENT_SOURCE_DIR} CACHE)
 
 add_custom_target(arc-check DEPENDS check-arc-mlir
-  COMMAND ${ARC_CARGO_BIN}/arc-cargo test --manifest-path=${ARC_SCRIPT_SRC_DIR}/Cargo.toml)
+  COMMAND cd ${ARC_SCRIPT_SRC_DIR}; cargo clean
```
Comment (frej) [inline arc-mlir/src/CMakeLists.txt:28]: This should be `arc-cargo clean` or else you won't clean the target dir you run `arc-cargo test` in. Otherwise lo

**Review by frej — APPROVED** [review]
Apart from the comment, this LGTM.

**Pull request merged**